### PR TITLE
[tooling] Add `dep_why` helper script to `dep_tree_resolver` tooling

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?

- Adds `dep_why.py` script which helps identify why a particular import in the
  dependency chain is pulled in.
- Adds a bit more info to the `dep_tree_resolver` README about caveats

### Motivation

While output of `dep_tree_resolver` is human-parsable, the file is usually too
large (100+ MB) to easily go through the file and identify the dependency chain
of the module of interest. This new script can be used to automate this process.

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

N/A

### Checklist

- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.

### CODE REVIEW

File: tools/dep_tree_resolver/dep_why.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    current_path = []
+    with open(dep_tree_filename) as infile:
+        for line in infile:
+            level, module_dep = parse_line(line)
+
+            if level <= len(current_path):
+                for _ in range(len(current_path) - level + 1):
+                    if len(current_path) > 0:
+                        current_path.pop()
+
```
Comment (mx-psi) [inline tools/dep_tree_resolver/dep_why.py:43]: I think this is equivalent, right?
```suggestion
                current_path = current_path[:level+1]
```

I find it easier to follow when written that way (but maybe that is just me)

Comment (sgnn7) [reply tools/dep_tree_resolver/dep_why.py]: Yeah that's much simpler and easier to follow! 

Fixed via https://github.com/DataDog/datadog-agent/pull/9308/commits/0985b2d57a3c059c25589cdf0fc0b56961eac6fd

File: tools/dep_tree_resolver/dep_why.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    print("Searching for '{}' in {}...".format(target_module, dep_tree_filename))
+
+    found = False
+    current_path = []
+    with open(dep_tree_filename) as infile:
+        for line in infile:
+            level, module_dep = parse_line(line)
+
+            if level <= len(current_path):
+                current_path = current_path[: level + 1]
```
Comment (mx-psi) [inline tools/dep_tree_resolver/dep_why.py:40]: You don't need the `len` check:
```python3
❯ python3
# ...
>>> a = [1,2,3,4,5]
>>> a[:4]
[1, 2, 3, 4]
>>> a[:200]
[1, 2, 3, 4, 5]
>>>
```

**Review by mx-psi — APPROVED** [review]
Looks good except for the nit above :) You will need to rebase to get all the pipeline fixes from the last days

**Pull request merged**